### DISCUSSION TITLE
Conditionally adding tw classes to standart html elements

### DISCUSSION THREAD

**Discussion by NetLancer** [ben-rogerson/twin.examples#87] [Q&A]
First, honest thanks for this great idea & tool, just getting into it, although i'm more of a newbie to nextjs too :)). 
The question is:  
Trying to rewrite an existing tailwind/nextjs template using tw tooling (tw/stitches) and came across the part to replace conditional styling. Based on examples found there i see two options: 
```
<div css={{...tw`fixed right-8 gap-3`, ...(${show} ? tw`md:flex` : tw`md:hidden`)}}> // show is an outside variable 
// OR 
<div css={[tw`fixed right-8 gap-3`, ${show} ? tw`md:flex` : tw`md:hidden`]}>
``` 
So, which one of the above lines would be the right one to use, or any other correct way to add conditional class(es) to the existing ones ? And if both are applicable, then what is the difference ? 
Pardon my a bit dummy question if it is, thanks a lot :)

**Comment by ben-rogerson** [comment]
Hey Ellie

```js
<div css={{...tw`fixed right-8 gap-3`, ...(show ? tw`md:flex` : tw`md:hidden`)}}>
```

Spreading conditional classes into an object requires a few extra characters than the array method below.
You'll see in your dev tools that the classes on the element are shallow merged.

```js
<div css={[tw`fixed right-8 gap-3`, show ? tw`md:flex` : tw`md:hidden`]}>
```

Using an array is clean and requires slightly less code and you won't see the objects merge in your dev tools, you'll see the base classes overwritten instead. 

But last time I tested, the array styling method had issues in Stitches. I added [the usage section in the stitches example readme](https://github.com/ben-rogerson/twin.examples/tree/master/next-stitches-typescript#usage) as it differed from the prop and styled component guides I'd previously added (which work with all the other supported libraries).

**Reply by NetLancer** [reply]
I see, so with stitches,  spreading method (former up top) comes more solid and tested, its fine. 
Then another question is: What is actually meant by "base classes"? So in ...tw`class1, class2` it seems to me some other pre-stylings considered (because of spread prefix) like some global css, upon which we're adding class1, class2 etc. 
This is what i thought to be the difference with using array method, which does not involve spread and thus only merges arrays without any other considerations. Am i right ?
